crates/ra_syntax/src/syntax_error.rs

```diff
@@ -4,14 +4,18 @@ use std::fmt;
 
 use ra_parser::ParseError;
 
-use crate::{validation::EscapeError, TextRange, TextUnit};
+use crate::{validation::EscapeError, TextRange, TextUnit, TokenizeError};
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct SyntaxError {
     kind: SyntaxErrorKind,
     location: Location,
 }
 
+// FIXME: Location should be just `Location(TextRange)`
+// TextUnit enum member just unnecessarily compicates things,
+// we should'n treat it specially, it just as a `TextRange { start: x, end: x + 1 }`
+// see `location_to_range()` in ra_ide/src/diagnostics
```

Comment:
I think this should be the target for the future refactoring. I am not entirely sure why we don't use just `TextRange` in as a location of `SyntaxError` instead. Or at least make location a newtype of `TextRange`? What do you think?

Reply:
Seems reasonable. In the end, when used to emit a diagnostic, it gets turned into a TextRange anyways: https://github.com/rust-analyzer/rust-analyzer/blob/d1330a4a65f0113c687716a5a679239af4df9c11/crates/ra_ide/src/diagnostics.rs#L118-L123

Reply:
Yeah, I just referred to that snippet in the comment)

Reply:
That you did :laughing: